daml-script/export/src/main/scala/com/daml/script/export/TreeUtils.scala

```diff
@@ -161,17 +163,35 @@ object TreeUtils {
         }
   }
 
-  case class CreatedContract(cid: ContractId, tplId: Identifier, path: List[Selector])
+  sealed trait CreatedContract {
```

Comment:
```suggestion
  sealed abstract class CreatedContract extends Product with Serializable {
```

Reply:
thanks for the suggestion @S11001001. I see that this also works, but I'd like to understand your reasoning. Googling for "trait vs abstract class" I see that traits are good for multiple inheritance (which I'm not using here) while abstract classes can take constructor parameters (which I'm also not using) and are supposed to be better for Java interop (which I'm also not using). Is it just a rule of thumb maybe?

Reply:
Instance of "use the least powerful construct that works".